pkg/endpoints/cluster.go

```diff
@@ -41,14 +44,23 @@ func (r Resource) ProxyRequest(request *restful.Request, response *restful.Respo
 		utils.RespondError(response, err, http.StatusNotFound)
 		return
 	}
+
 	uri := request.PathParameter("subpath") + "?" + parsedURL.RawQuery
 	forwardRequest := r.K8sClient.CoreV1().RESTClient().Verb(request.Request.Method).RequestURI(uri).Body(request.Request.Body)
 	forwardRequest.SetHeader("Content-Type", request.HeaderParameter("Content-Type"))
 	forwardResponse := forwardRequest.Do()
 
 	responseBody, requestError := forwardResponse.Raw()
 	if requestError != nil {
-		utils.RespondError(response, requestError, http.StatusNotFound)
+		errorInfo := string(responseBody)
+		errorInfo = strings.Replace(errorInfo, "\"", "", -1)
+		errorInfo = strings.Replace(errorInfo, "\\", "", -1)
+		errorInfo = strings.Replace(errorInfo, "\n", "", -1)
+		errorCodeString := strings.LastIndex(errorInfo, "code:")
+		errorCode := errorInfo[errorCodeString+5 : errorCodeString+8]
```

Comment:
This can go panic if we don't find the required content so make sure this is protected